static/js/unread.js

```diff
@@ -539,7 +539,12 @@ export function update_unread_topics(msg, event) {
 
 export function process_loaded_messages(messages) {
     for (const message of messages) {
-        if (message.unread) {
+        if (message.unread && !unread_messages.has(message.id)) {
+            // We only need to process it if the new message
+            // is an unread message that we already don't track.
+            // NOTE: This assumes we only process new messages using this function
+            // and don't handle any message edit updates.
+            any_untracked_unread_messages = true;
```

Comment:
It looks like the definition/return statements for this ended up in the next commit.